cnf-certification-test/accesscontrol/suite.go

```diff
@@ -303,3 +322,132 @@ func TestPodServiceAccount(env *provider.TestEnvironment) {
+
+		// Get any rolebindings that do not belong to the pod namespace.
+		roleBindings, err := rbTester.GetRoleBindings()
+		if err != nil {
+			failedPods = append(failedPods, put.Name)
+		}
+
+		if len(roleBindings) > 0 {
+			logrus.Warnf("Pod: %s has the following role bindings: %s", put.Name, roleBindings)
+			failedPods = append(failedPods, put.Name)
+		}
+	}
+	if n := len(failedPods); n > 0 {
+		logrus.Debugf("Pods with role bindings: %+v", failedPods)
+		ginkgo.Fail(fmt.Sprintf("%d pods have role bindings in other namespaces.", n))
+	}
+}
+
+// TestPodClusterRoleBindings verifies that the pod utilizes a valid cluster role binding that does not cross namespaces
+//nolint:dupl
+func TestPodClusterRoleBindings(env *provider.TestEnvironment) {
+	ginkgo.By("Should not have ClusterRoleBinding in other namespaces")
+	failedPods := []string{}
+
+	for _, put := range env.Pods {
+		ginkgo.By(fmt.Sprintf("Testing cluster role binding for pod: %s namespace: %s", put.Name, put.Namespace))
+		if put.Spec.ServiceAccountName == "" {
+			ginkgo.Skip("Can not test when serviceAccountName is empty. Please check previous tests for failures")
+		}
+
+		// Create a new object with the ability to gather clusterrolebinding specs.
+		rbTester := rbac.NewClusterRoleBindingTester(put.Spec.ServiceAccountName, put.Namespace, clientsholder.NewClientsHolder())
+
+		// Get any clusterrolebindings that do not belong to the pod namespace.
+		clusterRoleBindings, err := rbTester.GetClusterRoleBindings()
+		if err != nil {
+			failedPods = append(failedPods, put.Name)
+		}
+
+		if len(clusterRoleBindings) > 0 {
+			logrus.Warnf("Pod: %s has the following cluster role bindings: %s", put.Name, clusterRoleBindings)
+			failedPods = append(failedPods, put.Name)
+		}
+	}
+	if n := len(failedPods); n > 0 {
+		logrus.Debugf("Pods with cluster role bindings: %+v", failedPods)
+		ginkgo.Fail(fmt.Sprintf("%d pods have cluster role bindings in other namespaces.", n))
+	}
+}
+
+//nolint:funlen
+func TestAutomountServiceToken(env *provider.TestEnvironment) {
+	ginkgo.By("Should have automountServiceAccountToken set to false")
+
+	msg := []string{}
+	failedPods := []string{}
+	for _, put := range env.Pods {
+		ginkgo.By(fmt.Sprintf("check the existence of pod service account %s (ns= %s )", put.Namespace, put.Name))
+		gomega.Expect(put.Spec.ServiceAccountName).ToNot(gomega.BeEmpty())
+
+		// The token can be specified in the pod directly
+		// or it can be specified in the service account of the pod
+		// if no service account is configured, then the pod will use the configuration
+		// of the default service account in that namespace
+		// the token defined in the pod has takes precedence
+		// the test would pass iif token is explicitly set to false
+		// if the token is set to true in the pod, the test would fail right away
+		podServiceTokenStatus := rbac.DetermineStatus(put.Spec.AutomountServiceAccountToken)
```

Comment:
I find the DetermineStatus / Token status object function confusing. Why not just test on (same for sa) :
 - put.Spec.AutomountServiceAccountToken!=nil  -> there is no automountServiceAccountToken param configured
 - put.Spec.AutomountServiceAccountToken == true -> it is configured to true
 - put.Spec.AutomountServiceAccountToken == false -> it is configured to false

Reply:
Removed DetermineStatus().  Thanks!